### ISSUE TITLE
Spring persistence classes aren't supported

### ISSUE THREAD

**Issue by rjlasko** [liefke/org.fastnate#24]
Hello,

I recently tryed to use the Fastnate API to populate the contents of a DB (H2, created at app start with Flyway), and ran into a bug where it appears that the Fastnate API cannot generate values for generically defined IDs. 

My project's ORM leverages classes from the Spring API, which extend (after several levels) [AbstractPersistable.java](http://docs.spring.io/spring-data/jpa/docs/current/api/org/springframework/data/jpa/domain/AbstractPersistable.html) ([source](https://github.com/spring-projects/spring-data-jpa/blob/master/src/main/java/org/springframework/data/jpa/domain/AbstractPersistable.java)) as their `@MappedSuperclass`.

I found that since the ID is defined as the following,
```
@MappedSuperclass
public abstract class AbstractPersistable<PK extends Serializable> implements Persistable<PK> {
	@Id @GeneratedValue private PK id;
```

... the Fastnate IdGenerator [choked](https://github.com/liefke/org.fastnate/blob/master/fastnate-generator/src/main/java/org/fastnate/generator/context/IdGenerator.java#L66) when trying to determine the class type to be assigned.  This was my resulting error message:
```
org.fastnate.generator.context.ModelException: Can't handle number class for generated value: interface java.io.Serializable
	at org.fastnate.generator.context.IdGenerator.createNextValue(IdGenerator.java:66)
	at org.fastnate.generator.context.GeneratedIdProperty.addInsertExpression(GeneratedIdProperty.java:69)
	at org.fastnate.generator.EntitySqlGenerator.writeInserts(EntitySqlGenerator.java:203)
	at org.fastnate.generator.EntitySqlGenerator.write(EntitySqlGenerator.java:144)
	at org.fastnate.generator.EntitySqlGenerator.write(EntitySqlGenerator.java:118)
	at org.fastnate.generator.EntitySqlGenerator.write(EntitySqlGenerator.java:160)
	at org.fastnate.data.csv.AbstractCsvDataProvider.writeEntities(AbstractCsvDataProvider.java:333)
```

I believe that the underlying problem is [here](https://github.com/liefke/org.fastnate/blob/master/fastnate-generator/src/main/java/org/fastnate/generator/context/EntityClass.java#L364).  It appears that while this function successfully identifies the class from which the `@Id` is declared, the process does not sustain the generics that truly define the ID's type for a concrete class that would extend it, causing the code to think that the ID is of type `Serializable`, when it should be `Long` (as in my case).

**Comment by liefke** [comment]
Thank you for reporting this issue. I've never tested _Spring Data JPA_ with _Fastnate_ and wasn't aware that it is possible to define any mapped property with a generic type.

Your analysis of the problem is quite right, but I decided to fix the problem for all properties (not only IDs). I've added a test case as well for `AbstractPersistable`. You can find the details in the commit above.

This fix is included in 1.4.0 - if you want to test it before that release you can use our [snapshot builds](https://fastnate.org/#download).

**Comment by rjlasko** [comment]
Excellent!  I'm glad that you went with the all-inclusive approach.
